### PR TITLE
Fix URLs used in stable docs deploys

### PR DESCRIPTION
**Pull request by Eric-Arellano**
I missed this in https://github.com/Qiskit/qiskit-metapackage/pull/1740. We renamed the option from `content_prefix` to `docs_url_prefix`.

### CODE REVIEW

File: tools/deploy_documentation.sh [hunk 1]
```diff
@@ -21,7 +21,7 @@ sudo apt-get install -y ./rclone.deb
 RCLONE_CONFIG_PATH=$(rclone config file | tail -1)
 
 # Build the documentation.
-tox -edocs -- -D content_prefix=documentation -j auto
+tox -edocs -- -j auto
```
Comment (Eric-Arellano) [inline tools/deploy_documentation.sh:24]: Now set here, so no need to configure via CLI override:

https://github.com/Qiskit/qiskit-metapackage/blob/86a29b4b4b744417c9f478d2252aa735db8bd568/docs/conf.py#L43

**Review by jakelishman — DISMISSED** [review]
Seems odd to me to have this inconsistency in where the documentation prefix is defined ("as an input to sphinx-build" seemed more logical than "universal conf file" to me, since it's inherently tied to one particular deployment), but that's not a particularly big deal.  Good to get the options fixed.

**Comment by Eric-Arellano** [comment]
> Seems odd to me to have this inconsistency in where the documentation prefix is defined ("as an input to sphinx-build" seemed more logical than "universal conf file" to me, since it's inherently tied to one particular deployment)

I'd be fine with either way. 

The important detail is that qiskit_sphinx_theme will error if versions or translations are used and `docs_url_prefix` isn't set. So, I suppose then it would be safer to not set it in `conf.py` because it forces every call site of `sphinx-build` to reason about it, rather than not realizing they're using the default.

File: docs/versionutils.py [hunk 2]
```diff
@@ -25,6 +25,7 @@
 
 
 def setup(app):
+    app.connect("config-inited", _extend_html_context)
```
Comment (Eric-Arellano) [inline docs/versionutils.py:28]: My bad. This results in the Previous Releases feature not showing up.

**Review by HuangJunye — APPROVED** [review]

**Pull request merged**